src/tc-yaml.js

```diff
@@ -0,0 +1,234 @@
+const jparam = require('json-parameterization');
+const _ = require('lodash');
+const slugid = require('slugid');
+const jsone = require('json-e');
+const tc = require('taskcluster-client');
+
+// Assert that only scope-valid characters are in branches
+const branchTest = /^[\x20-\x7e]*$/;
+
+/**
+ * Get scopes and attach them to the task.
+ * v1 function
+ */
+function createScopes(config, payload) {
```

Comment:
This function and the next probably belong in the VersionOne and VersionZero classes.